test/language/expressions/assignment/destructuring/obj-prop-nested-array-yield-expr.js

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: >
+    When a `yield` token appears within the Initializer of a nested
+    destructuring assignment and within a generator function body, it should
+    behave as a YieldExpression.
+es6id: 12.14.5.4
```

Comment:
having words that are sometimes magic and sometimes not magic -> refactoring hazard that folks have generally tried to avoid. It's going to be another one of those things that bites people :(